### PR TITLE
Group and explain modbus climate entity settings

### PR DESCRIPTION
**Pull request by rutgerrademaker**
## Proposed change

With this commit I'm trying to group the different "parts" when configuring a modbus climate entity, being:

- General settings like C or F and the temperature steps
- Read current temperature
- Set target temperature
- Set/Read HVAC mode
- Set/Read fan mode
- Set/Read swing mode
- Set/read on off state

I think this is needed as in the current state, personally, I found the naming of the config settings to be quite confusing. 
Only by reading the actual Python code it became clear how I could fully configure my AC units.
By grouping the settings and explicitly name which "part" some setting applies to I think more people will be able to configure their AC units.

This contribution was sparked by the fact I'm trying to add a feature to the modbus climate entity without introducing backwards compatible changes. By improving the docs I'm hoping to pave the way for getting that PR merged

https://github.com/home-assistant/home-assistant.io/pull/33741
https://github.com/home-assistant/core/pull/121680

I'd like this PR to be judged on at least these things:

1) Would grouping/shuffling the current entities indeed make it more readable?
2) Did I correctly reverse-engineer the current settings?
3) Not sure If I'm allowed to use yaml comments in order to separate the "groups" of settings (an/or if there is a better way to do this)

N.B. somewhere down the road I think it would make sense to propose some modbus climatev2 entity in which more flexibility and declarative naming for different settings could be introduced.

PR's like https://github.com/home-assistant/core/pull/118013 also show that more flexibility might be needed in order to control different units. Until then though, let's just improve what we have without having to break backwards compatibility

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [x] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

## Summary by CodeRabbit

- **New Features**
	- Expanded configuration options for managing climate entities, including new parameters such as temperature unit, precision, temperature step, maximum temperature, minimum temperature, input type, offset, target temperature registers, scale, structure, and swap.
	- Enhanced clarity on controlling HVAC state with updates to the HVAC On/Off register, fan mode register, and swing mode register.
- **Documentation**
	- Improved descriptions and usability of existing parameters related to climate control systems.

### CODE REVIEW

File: source/_integrations/modbus.markdown [hunk 1]
```diff
@@ -722,86 +793,18 @@ climates:
...
-      default: C
-      type: list
-      keys:
-        C:
-          description: "Celsius"
-        F:
-          description: "Fahrenheit"
     write_registers:
-      description: "Request type, use `write_registers` if true  else `write_register`."
+      description: "If `true` use `write_registers` to control the On/Off state (`hvac_onoff_register`), else use `write_register`."
```
Comment (c0ffeeca7) [inline source/_integrations/modbus.markdown]:
```suggestion
      description: "If `true` use `write_registers` to control the On/Off state (`hvac_onoff_register`), else use `write_register`.
```

File: source/_integrations/modbus.markdown
Comment (rutgerrademaker) [inline source/_integrations/modbus.markdown:1]: Thanks! missed that (was still struggling to get all the dev dependencies / linting installed locally)

File: source/_integrations/modbus.markdown [hunk 2]
```diff
@@ -530,7 +530,7 @@ The master configuration like device_class are automatically copied to the slave
 
 ## Configuring climate entities
 
-The Modbus climate platform allows you to monitor a thermostat or heaters as well as set a target temperature, HVAC mode and fan state.
+The Modbus climate platform allows you to monitor a thermostat or heaters as well as set a target temperature, HVAC mode, swing mode and fan state.
```
Comment (c0ffeeca7) [inline source/_integrations/modbus.markdown]:
```suggestion
The Modbus climate platform allows you to monitor a thermostat or heaters as well as set a target temperature, HVAC mode, swing mode, and fan state.
```

File: source/_integrations/modbus.markdown [hunk 3]
```diff
@@ -577,6 +607,54 @@ climates:
...
+          description: "Holding register."
+        input:
+          description: "Input register."
+    offset:
+      description: "Final offset for current temperature (output = scale * value + offset)."
+      required: false
+      type: float
+      default: 0
+    target_temp_register:
+      description: "Register address for target temperature (Setpoint). Using a list, it is possible to define one register for each of the available HVAC Modes. The list has to have a fixed size of 7 registers corresponding to the 7 available HVAC Modes, as follows: Register **1: HVAC AUTO mode**; Register **2: HVAC Cool mode**; Register **3: HVAC Dry mode**; Register **4: HVAC Fan only mode**; Register **5: HVAC Heat mode**; Register **6: HVAC Heat Cool mode**; Register **7: HVAC OFF mode**. It is possible to set duplicated values for the modes where the devices has not a related register."
```
Comment (c0ffeeca7) [inline source/_integrations/modbus.markdown]:
```suggestion
      description: "Register address for target temperature (Setpoint). Using a list, it is possible to define one register for each of the available HVAC Modes. The list has to have a fixed size of 7 registers corresponding to the 7 available HVAC Modes, as follows: Register **1: HVAC AUTO mode**; Register **2: HVAC Cool mode**; Register **3: HVAC Dry mode**; Register **4: HVAC Fan only mode**; Register **5: HVAC Heat mode**; Register **6: HVAC Heat Cool mode**; Register **7: HVAC OFF mode**. It is possible to set duplicated values for the modes where the devices don't have a related register."
```

File: source/_integrations/modbus.markdown [hunk 4]
```diff
@@ -722,86 +793,18 @@ climates:
...
-      type: list
-      keys:
-        C:
-          description: "Celsius"
-        F:
-          description: "Fahrenheit"
     write_registers:
-      description: "Request type, use `write_registers` if true  else `write_register`."
+      description: "If `true` use `write_registers` to control the On/Off state (`hvac_onoff_register`), else use `write_register`.
+      Note that is not yet possible to control the On/Off state via a coil."
```
Comment (c0ffeeca7) [inline source/_integrations/modbus.markdown]:
```suggestion
      Note that it is not yet possible to control the On/Off state via a coil."
```

**Comment by c0ffeeca7** [comment]
I saw it a bit late, but this PR needs to be pointed against the `next` branch. Please rebase.

**Comment by rutgerrademaker** [comment]
> I saw it a bit late, but this PR needs to be pointed against the `next` branch. Please rebase.

Are you sure, there's nothing new in this change, according to the guidelines it should be targeting the current branch

> Type of change
> 
> Spelling, grammar or other readability improvements (current branch).
> Adjusted missing or incorrect information in the current documentation (current branch).

**Comment by c0ffeeca7** [comment]
> > I saw it a bit late, but this PR needs to be pointed against the `next` branch. Please rebase.
> 
> Are you sure, there's nothing new in this change, according to the guidelines it should be targeting the current branch
> 
> > Type of change
> > Spelling, grammar or other readability improvements (current branch).
> > Adjusted missing or incorrect information in the current documentation (current branch).

You are right, I was confused. As the core PR (https://github.com/home-assistant/home-assistant.io/pull/33903#issuecomment-2247611142) is referenced in the PR description, I thought this was the parent.

**Comment by rutgerrademaker** [comment]
@c0ffeeca7 @janiversen is there anything left I should address?

File: source/_integrations/modbus.markdown [hunk 5]
```diff
@@ -540,13 +540,43 @@ climates:
   required: false
   type: map
   keys:
+    temperature_unit:
+      description: "Temperature unit, C or F."
```
Comment (frenck) [inline source/_integrations/modbus.markdown]:
```suggestion
      description: "Temperature unit: C or F."
```

**Review by frenck — APPROVED** [review]
Thanks, @rutgerrademaker 👍 

../Frenck

**Pull request merged**